### PR TITLE
use GitHub Actions

### PR DESCRIPTION
**Pull request by amontoison**
close #9

### CODE REVIEW

**Comment by dpo** [comment]
Wow. What's all that GenericLinearAlgebra junk? How is that getting pulled in?

I added keys to push the docs but it looks like it's complaining that `ssmc_matrices` has no docstring. It does though...

**Comment by dpo** [comment]
You may be able to fix CirrusCI builds as in #9.

**Comment by amontoison** [comment]
> Wow. What's all that GenericLinearAlgebra junk? How is that getting pulled in?

JuliaDB is the culprit.

> I added keys to push the docs but it looks like it's complaining that `ssmc_matrices` has no docstring. It does though...

We forgot to add it in ` docs/src/api.md`.

**Comment by dpo** [comment]
Hmm. Windows failures say
```
tar: D\:\a\\SuiteSparseMatrixCollection.jl\\SuiteSparseMatrixCollection.jl\\src\\..\\data\\MM\\Oberwolfach: Cannot open: No such file or directory
103
tar: Error is not recoverable: exiting now
104
ERROR: LoadError: failed process: Process(`tar --force-local -zxf 'D:\a\SuiteSparseMatrixCollection.jl\SuiteSparseMatrixCollection.jl\src\..\data\MM\Oberwolfach\LF10.tar.gz' -C 'D:\a\SuiteSparseMatrixCollection.jl\SuiteSparseMatrixCollection.jl\src\..\data\MM\Oberwolfach'`, ProcessExited(2)) [2]
```
In the future, we'll have to use artifacts and get rid of JuliaDB...

In the meantime, maybe https://github.com/JuliaIO/Tar.jl ?

**Comment by amontoison** [comment]
https://discourse.julialang.org/t/cross-platform-unpacking-of-tar-archive/31459
https://github.com/oxinabox/DataDeps.jl/blob/master/src/post_fetch_helpers.jl

It's also true for 7z...
tar

**Comment by dpo** [comment]
This package is going to move artifacts + dataframes pretty soon...

File: Project.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
 JuliaDB = "a93385a2-3734-596a-9a66-3cfbb77141e6"
 
 [compat]
-JuliaDB = "^0.12"
+JuliaDB = "^0.13"
```
Comment (dpo) [inline Project.toml:10]:
```suggestion
JuliaDB = "^0.12"
```
Let's not bother doing this. I have a PR to phase out JuliaDB.

File: docs/Project.toml [hunk 2]
```diff
@@ -3,5 +3,5 @@ Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
 JuliaDB = "a93385a2-3734-596a-9a66-3cfbb77141e6"
 
 [compat]
-Documenter = "~0.21"
-JuliaDB = "^0.12"
+Documenter = "~0.25"
+JuliaDB = "^0.13"
```
Comment (dpo) [inline docs/Project.toml:7]:
```suggestion
JuliaDB = "^0.12"
```

File: src/SuiteSparseMatrixCollection.jl [hunk 3]
```diff
@@ -114,8 +114,12 @@ function fetch_ssmc(matrices; format="MM")
...
     end
     if ext == "tar.gz"
-      tar_options = Sys.iswindows() ? "--force-local" : ""
-      run(`tar $tar_options -zxf $fname -C $g_path`)
+      if Sys.iswindows()
+        exe7z = isdefined(Base, :LIBEXECDIR) ? joinpath(Sys.BINDIR, Base.LIBEXECDIR, "7z.exe") : joinpath(Sys.BINDIR, "7z.exe")
+        pipeline(`$exe7z x $fname -y -so`, `$exe7z x -si -y -ttar -o$g_path`)
+      else
+        run(`tar -zxf $fname -C $g_path`)
+      end
```
Comment (dpo) [inline src/SuiteSparseMatrixCollection.jl:122]: Let's not do this either. It looks like it's the tar file that has some issues, not the tar command. I'll see if I can make those matrices available as artifacts instead.

**Review by dpo — CHANGES_REQUESTED** [review]
Rebase on top of #12 when it's merged.

**Comment by dpo** [comment]
It turns out that the tarballs themselves are malformed. Tar.jl can't read them. So artifacts won't work either. What I suggest is that in the tests, if the system is Windows, we download the MAT format.

**Comment by dpo** [comment]
Phew!!! Awesome, thank you!

**Pull request merged**